src/main/java/ua/lviv/iot/parkingServer/datastorage/AbstractFIleStore.kt

```diff
@@ -0,0 +1,59 @@
+package ua.lviv.iot.parkingServer.datastorage
+
+import ua.lviv.iot.parkingServer.model.CsvData
+import ua.lviv.iot.parkingServer.utils.Util
+import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStreamWriter
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.*
+
+import java.util.concurrent.atomic.AtomicLong
+import kotlin.math.max
+
+abstract class AbstractFileStore<T : CsvData> {
+    private fun readRecords(): List<T> {
+        val fileName = "$RESULT_FOLDER/${getRecordName()}-${Util.getDateTodayInString()}"
+        return if (Files.exists(Paths.get(fileName))) readRecordsFrom(File(fileName)) else emptyList()
+    }
+
+    private fun readRecordsFrom(csv: File): List<T> {
+        val result: MutableList<T> = LinkedList()
+        val scanner = Scanner(csv, StandardCharsets.UTF_8)
+        if (scanner.hasNextLine()) scanner.nextLine()
+        while (scanner.hasNextLine()) {
+            val values: List<String> = scanner.nextLine().split(", ").toList()
+            result.add(convert(values))
+        }
+        return result
```

Comment:
Alternative, more kotlinish approach:
```kotlin
private fun readRecordsFrom(csv: File): List<String> {
    return csv.useLines { sequence -> // useLines - the function reads a file lines and provides sequesnce of lines. 
                                                           // automatically closes the file
        sequence // Consider sequence as Stream in Java                 
                .drop(1) // skip the first line. 
                .map { line -> line.split(", ") }
                .map { lineItems -> convert(lineItems) }
                .toList()
    }
}
```

Reply:
The approach above does exactly the same your code does. But it less verbose and more elegant, isn't it?

In general I'm ok with your solution(excepts unclosed resource) but it's more java-like